crates/spidermonkey-embedding-splicer/src/bindgen.rs

```diff
@@ -1043,6 +1054,11 @@ impl EsmBindgen {
                 ");
         }
         for (export_name, binding) in &self.exports {
+            if export_name == "wasi:http/incoming-handler@0.2.0"
+                && !guest_exports.contains(&"incomingHandler".to_string())
+            {
+                continue;
+            }
```

Comment:
I think this should be a more general mechanism, where we try to find a JS export for `export_name` as currently. And if that fails, we check if the input wasm file already has the export, and keep it if so. Otherwise, propagate the error as per usual.

I don't know the code enough to say exactly how best to implement it, but the logic in pseudo-code would roughly be:
```rust
if try_finding_export_for(export_name).is_err() {
  // JS content doesn't have an export for `export_name`.
  if input_wasm_already_has_export(export_name) {
    // If the input wasm file already contains this export, ignore the error.
    continue;
  }
  // Otherwise, propagate the error as before.
```

This all would then have to be used in the splicing code as well, of course.

Reply:
I also think I need to figure out how to go from a JS export name to the `wit` export.